### PR TITLE
Do forge authentication in one function

### PR DESCRIPTION
**Pull request by fthomas**
This adds `ForgeSelection.authenticate` which modifies requests for authentication for a given forge type. This is nicer than to scatter this logic over multiple files and I think it will also be useful in fixing #1468.

### CODE REVIEW

**Review by exoego — APPROVED** [review]

File: modules/core/src/main/scala/org/scalasteward/core/forge/ForgeSelection.scala [hunk 1]
```diff
@@ -30,57 +32,43 @@ import org.scalasteward.core.util.HttpJsonClient
...
+      case Bitbucket       => _.putHeaders(basicAuth(user)).pure[F]
+      case BitbucketServer =>
+        // Bypass the server-side XSRF check, see
+        // https://github.com/scala-steward-org/scala-steward/pull/1863#issuecomment-754538364
+        val xAtlassianToken = Header.Raw(ci"X-Atlassian-Token", "no-check")
+        _.putHeaders(basicAuth(user), xAtlassianToken).pure[F]
+      case GitHub => _.putHeaders(basicAuth(user)).pure[F]
+      case GitLab => _.putHeaders(Header.Raw(ci"Private-Token", user.accessToken)).pure[F]
     }
+
```
Comment (mzuehlke) [inline modules/core/src/main/scala/org/scalasteward/core/forge/ForgeSelection.scala:71]: what about:
```Scala
case BitbucketServer => _.putHeaders(basicAuth(user), xAtlassianToken).pure[F]

...

// Bypass the server-side XSRF check, see
// https://github.com/scala-steward-org/scala-steward/pull/1863#issuecomment-754538364
private val xAtlassianToken = Header.Raw(ci"X-Atlassian-Token", "no-check")
```

to make the patter matching simpler

Comment (mzuehlke) [reply modules/core/src/main/scala/org/scalasteward/core/forge/ForgeSelection.scala:71]: And I like the overall code reduction 😄 👍

Comment (fthomas) [reply modules/core/src/main/scala/org/scalasteward/core/forge/ForgeSelection.scala:71]: Good suggestion! That nicely aligns all cases.

Comment (fthomas) [reply modules/core/src/main/scala/org/scalasteward/core/forge/ForgeSelection.scala:71]: Done in https://github.com/scala-steward-org/scala-steward/pull/2920/commits/da4ec126317ab27885cd7dfb0e32f8b0605241c5.

**Pull request merged**